site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -47,7 +47,87 @@ private function validateDiffViewerType(string $type) {
         }
         return true;
     }
-
+    /* Used in ajaxRandomizePeets as a helper function.
+    @param  $type
+    */
+    public function arr_rotate($array,$rotate_count) {
+        
+    }
+/**
+        * Route for randomizing peer assignments
+        * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/ajaxRandomizePeers", methods={"POST"})
+        * @AccessControl(role="INSTRUCTOR")
+    */ 
+    public function ajaxRandomizePeers($gradeable_id){
+        $number_to_grade=$_POST['number_to_grade'];
+       // $gradeable_id=$_POST['gradeable_id'];
+        $gradeable = $this->tryGetGradeable($gradeable_id);
+         if ($gradeable === false) {
+             $this->core->addErrorMessage('Invalid Gradeable!');
+             $this->core->redirect($this->core->buildCourseUrl());
+         }
+         $order = new GradingOrder($this->core, $gradeable, $this->core->getUser(), true);
+         $student_array= [];
+         $student_list= [];
+         $students = $this->core->getQueries()->getUsersByRegistrationSections($order->getSectionNames());
+         foreach ($students as $student) {
+              $reg_sec = ($student->getRegistrationSection() === null) ? 'NULL' : $student->getRegistrationSection();
+              $sorted_students[$reg_sec][] = $student;
+              array_push($student_list, [
+                  'user_id' => $student->getId(),
+                 // reg_section => $reg_sec,
+              ]);
+              array_push($student_array,$student->getId());
+          }
+          $graded_array=$student_array;
+          $final_info=[];
+          shuffle($student_array);
+          array_push($final_info,$student_array);
+          $final_grading_info=[];
+          $max_offset=sizeof($student_array);
+          $offset_array=[];
+          for($i=0;$i<$number_to_grade;++$i){
+            $random_offset=rand(1,$max_offset);
+            array_push($offset_array,$random_offset);
+          }
+          foreach($offset_array as $element){
+            $temp_array=$student_array;
+            for ($i = 0; $i < $element; $i++) {
+                array_push($temp_array, array_shift($temp_array));
+            }
+            //arr_rotate($temp_array,);
+            array_push($final_info,$temp_array);
+          }
+          for($i=0;$i<sizeof($final_info[0]);++$i){
+              $temp=[];
+              for($j=1;$j<sizeof($final_info);++$j){
+                  array_push($temp,$final_info[$j][$i]);
+              }
```

Comment:
Add comments here to explain what's going on. This will be important if a new dev ever has to touch this code.

Reply:
Updated!